src/services/TechRecordsService.ts

```diff
@@ -297,6 +297,33 @@ class TechRecordsService {
         throw new HTTPError(500, HTTPRESPONSE.INTERNAL_SERVER_ERROR);
       });
   }
+
+  public async updateTechRecordStatusCode(vin: string, newStatus: STATUS = STATUS.CURRENT) {
+    const techRecordWrapper: ITechRecordWrapper = await this.getTechRecordsList(vin, STATUS.ALL);
+    const provisionalTechRecords = techRecordWrapper.techRecord.filter((techRecord) => techRecord.statusCode === STATUS.PROVISIONAL);
```

Comment:
So update can only EVER be from provisional to ... something? Never goes back?

Reply:
At the moment yes. The provisional tech-record will become `archived` and I will add a new `current` tech-record to the array. The next time I will search for this vehicle, it won't be provisional anymore.
In the future I'm not sure if the status will be changed to something else (from VTM for example). That's why I provided the `newStatus` parameter which is set to default `STATUS.CURRENT` but can be passed as a queryString parameter to the lambda function.

Reply:
More concerned about the situation where a Current record might need to be archived

Reply:
Apparently not a current concern